crates/autopilot/src/infra/solvers/dto/solve.rs

```diff
@@ -167,19 +169,82 @@ pub enum Side {
 #[serde_as]
 #[derive(Clone, Debug, Deserialize)]
 #[serde(rename_all = "camelCase", deny_unknown_fields)]
-pub struct Solution {
-    /// Unique ID of the solution (per driver competition), used to identify
-    /// it in subsequent requests (reveal, settle).
-    #[serde_as(as = "serde_with::DisplayFromStr")]
-    pub solution_id: u64,
-    #[serde_as(as = "HexOrDecimalU256")]
-    pub score: U256,
-    /// Address used by the driver to submit the settlement onchain.
-    pub submission_address: H160,
-    pub orders: HashMap<boundary::OrderUid, TradedOrder>,
-    #[serde_as(as = "HashMap<_, HexOrDecimalU256>")]
-    pub clearing_prices: HashMap<H160, U256>,
-    pub gas: Option<u64>,
+#[serde(untagged)]
+pub enum Solution {
```

Comment:
Why should this migration require an enum here?
We could simply have deserialization logic for the `id` field that is able to parse `u64` and `strings`.

Reply:
Sure, added.

I was too lazy to implement the custom deserializator for a temporary code.